sparse/_utils.py

```diff
@@ -140,9 +145,23 @@ def random(
     # See https://github.com/scipy/scipy/blob/master/LICENSE.txt
     from ._coo import COO
 
+    if density is not None and nnz is not None:
+        raise ValueError("'density' and 'nnz' are mutually exclusive")
+
+    if density is None:
+        density = 0.01
+    if not (0 <= density <= 1):
+        raise ValueError("density {} is not in the unit interval".format(density))
```

Comment:
Can you test this error as well, because it's new?

Reply:
It is checked by `test_random_invalid_density_and_nnz` with `density=1.01` and `density=-0.01`.